packages/engine/src/scene/systems/ShadowSystem.ts

```diff
@@ -96,7 +116,73 @@ export default async function ShadowSystem(world: World) {
     return null
   })
 
+  const shadowComponentQuery = defineQuery([DropShadowComponent])
+
+  const shadowOffset = new Vector3(0, 0.01, 0)
+  const shadowGeometry = new PlaneGeometry(1, 1, 1, 1)
+  const loader = new TextureLoader()
+  const texture = loader.load(
+    (process.env['VITE_FILE_SERVER'] ?? 'https://localhost:8642') + '/projects/default-project/public/drop-shadow.png'
+  )
+  const shadowMaterial = new MeshBasicMaterial({ map: texture, side: DoubleSide, transparent: true })
+  let dropShadows = new InstancedMesh(shadowGeometry, shadowMaterial, shadowComponentQuery().length)
+
+  let sceneObjects = Array.from(Engine.instance.currentWorld.objectLayerList[ObjectLayers.Camera] || [])
+  console.log(sceneObjects)
+
+  const CreateDropShadows = () => {
```

Comment:
this can be refactored into a query reactor, in which if the conditions of regular shadows disabled & casts shadows are met on an entity, we create a drop shadow